### PR TITLE
Logger

### PR DESCRIPTION
**Pull request by mpclarkson**
Logger branch

### CODE REVIEW

**Comment by tannernelson** [comment]
Looks great. If the status checks pass I will merge it.

File: Sources/Request.swift [hunk 1]
```diff
@@ -69,6 +69,9 @@ public class Request {
         } else {
             self.data = Request.parseQueryData(path)
         }
+        
+        //Testing logs
+        Log.info("Received \(method.rawValue) request for \(path)")
```
Comment (shnhrrsn) [inline Sources/Request.swift]: This seems like it should be a lower log level, perhaps verbose or debug?

**Comment by mpclarkson** [comment]
Ok, I'll write some tests for it tonight. Also about to submit another PR to update the travis config and also run `swift build` as part of the ci process. 

Anyone have any ideas how to run tests with the SPM? We can perhaps do a second build using Xcode.

**Comment by tannernelson** [comment]
I don't think there is anyway to run tests in pure SPM right now. It's coming soon. You have to run them through Xcode though which is hard to do on Travis with Swift 2.2.

It's definitely feasible to make sure to police each other and verify we are running the tests before merging, though.

File: Sources/Request.swift (hunk 1 above)
Comment (tannernelson) [inline Sources/Request.swift]: Agreed.

Comment (tannernelson) [inline Sources/Request.swift]: Oh wait, sorry, I misread the code. Actually I think `info` could be good for this. I've seen a lot of web frameworks echo this information. PHP and Node both do it I think.

Comment (shnhrrsn) [inline Sources/Request.swift]: I think it's valuable, I'm just not sure it should be at an "info" level.  That's a lot of noise on a production deploy for logs, which would mean you'd need to do all real logging above info if you want to filter it out.

Comment (mpclarkson) [inline Sources/Request.swift]: Yes definitely... Sorry, that was just for testing on my branch.

**Comment by mpclarkson** [comment]
I've changed the example log in the request t o `VERBOSE`

**Comment by tannernelson** [comment]
@mpclarkson Let me know when the tests are done. Also, if you could prepare a `Log.md` page for the Wiki I will add that when this is pushed. It will be release `0.1.10`

**Comment by mpclarkson** [comment]
@tannernelson yes will add tests later today

**Comment by mpclarkson** [comment]
@tannernelson tests are written and pushed. Should be fine to merge.

I'll do the Log.md a bit later... I have some 'real' work to get done now.

**Comment by tannernelson** [comment]
Hmm... looks like it thinks there's a merge conflict. Try merging master back into your log branch.

I forgot this wasn't merged and was just trying to use it haha.

**Comment by mpclarkson** [comment]
yeah... there's two. One is in the Xcode project which is a pain.. working on it.

**Pull request closed without merging**